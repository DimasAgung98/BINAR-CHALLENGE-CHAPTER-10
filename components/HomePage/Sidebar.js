import React from 'react';
import Link from 'next/link';

function Sidebar() {
    return (
        <>
            <div className='container-fluid bg-light'>
                <div className='row'>
                    <div className='col-12 list-menu'>
                        <Link href='/home'><a><li>HOME</li></a></Link>
                        <Link href='/game/rock-paper-scissors'><li><a>RPS</a></li></Link>
                        <Link href='/game/gta'><a><li>GTA V</li></a></Link>
                        <Link href='/game/fall-guys'><a><li>FALLGUYS</li></a></Link>
                        <Link href='/game/dota'><a><li>DOTA 2</li></a></Link>
                        <Link href='/game/watch-dogs'><a><li>WATCHDOGS</li></a></Link>
                        <Link href='/game/valorant'><a><li>VALORANT</li></a></Link>
                        <Link href='/game/cyber-punk'><a><li>CYBERPUNK</li></a></Link>
                    </div>
                </div>
            </div>
        </>
    )
}

export default Sidebar