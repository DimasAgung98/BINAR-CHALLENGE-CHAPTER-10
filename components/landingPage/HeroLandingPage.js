import React from 'react';
import Image from 'next/image';
import Link from 'next/link';
//IMPORT REACTSTRAP COMPONENTS
import { Button } from 'reactstrap';
//IMPORT COMPONENTS PAGE
import CarouselSection from '../landingPage/Carousel';
import OurGames from '../global/OurGames';
//IMPORT ICONS
import { BiUser, BiIdCard, BiBarChartAlt, BiFace, BiGitMerge, BiJoystick } from 'react-icons/bi'
//IMPORT LOGO IMAGES
import Man1 from '../../public/man3.jpg';
import Man2 from '../../public/man4.png';
import Man3 from '../../public/man5.jpg';
//IMPORT VIDEO
import BG from '../../public/BG.mp4';


function HeroLandingPage() {
  return (
    <>
      {/* 1st page */}
      <div className='container-fluid'>
        <div className='back-video'>
          <video className='resp-video' autoPlay muted source src={BG} loop />
        </div>
        <div className='row main-section'>
          <div className='col-lg-12'>
            <div className='title-text'>
              <h1>TEAM<span>ONE</span></h1>
              <p>Play various games from Team One Project and enjoy the fun. <br></br>Hurry up register your account and play the game</p>
            </div>
            <div className='button-main text-center'>
              <Link href='/register'><Button color='warning' outline className='button-reg'>REGISTER NOW</Button></Link>
            </div>
          </div>
        </div>
      </div>
      {/* 2nd page */}
      <CarouselSection />
      <div className='container-fluid bg-light ptb2'>
        <div className='row px-5'>
          <div className='col-lg-4 col-md-6 col-sm-6 col-12 text-left'>
            <div className='text-card'>
              <div className='card card-styles'>
                <div className='card-body'>
                  <BiUser className='card-icon ms-3' />
                  <h5 className='card-title ms-3'>JOINED USER</h5>
                  <p className='ms-3 card-desc'> More than 1000 players join the game and play together</p>
                </div>
              </div>
            </div>
          </div>
          <div className='col-lg-4 col-md-6 col-sm-6 col-12 text-left'>
            <div className='text-card'>
              <div className='card card-styles'>
                <div className='card-body'>
                  <BiJoystick className='card-icon ms-3' />
                  <h5 className='card-title ms-3'>MANY GAME</h5>
                  <p className='ms-3 card-desc'> There are several choices of interesting games from all genres that can be played</p>
                </div>
              </div>
            </div>
          </div>
          <div className='col-lg-4 col-md-6 col-sm-6 col-12 text-left'>
            <div className='text-card'>
              <div className='card card-styles'>
                <div className='card-body'>
                  <BiBarChartAlt className='card-icon ms-3' />
                  <h5 className='card-title ms-3'>LEADERBOARD SYSTEM</h5>
                  <p className='ms-3 card-desc'> There is a leaderboard feature to compete with other players</p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className='row px-5 pt-5'>
          <div className='col-lg-4 col-md-6 col-sm-6 col-12 text-left'>
            <div className='text-card'>
              <div className='card card-styles'>
                <div className='card-body'>
                  <BiFace className='card-icon ms-3' />
                  <h5 className='card-title ms-3'>24 HOURS SUPPORT</h5>
                  <p className='ms-3 card-desc'>We are all ready to help you 24 hours if you need help</p>
                </div>
              </div>
            </div>
          </div>
          <div className='col-lg-4 col-md-6 col-sm-6 col-12 text-left'>
            <div className='text-card'>
              <div className='card card-styles'>
                <div className='card-body'>
                  <BiGitMerge className='card-icon ms-3' />
                  <h5 className='card-title ms-3'>CONNECT SOCIAL MEDIA</h5>
                  <p className='ms-3 card-desc'> The game you play can connect with your social media and can share your game and point</p>
                </div>
              </div>
            </div>
          </div>
          <div className='col-lg-4 col-md-6 col-sm-6 col-12 text-left'>
            <div className='text-card'>
              <div className='card card-styles'>
                <div className='card-body'>
                  <BiIdCard className='card-icon ms-3' />
                  <h5 className='card-title ms-3'>SAFE PRIVACY</h5>
                  <p className='ms-3 card-desc'> Your privacy data will be safe with us so you can keep playing and have fun</p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      {/* 3rd page */}
      <OurGames />
      {/* 4th page */}
      <div className='container-fluid py-5 bg-light'>
        <div className='row py-5 px-5 text-center'>
          <h5 className='fw-bold testi'>TESTIMONIALS</h5>
          <h2> What People Say</h2>
        </div>
        <div className='row px-5'>
          <div className='col-md-4'>
            <div className="card-testi">
              <Image src={Man1} className="card-img-top" alt="man" />
              <div className="card-body mt-3">
                <h4 className='name-testi'>BRODY</h4>
                <p className="card-text text-center">&quot;The games here are very fun, I play until I lose track of time&quot;</p>
              </div>
            </div>
          </div>
          <div className='col-md-4'>
            <div className="card-testi">
              <Image src={Man3} className="card-img-top" alt="man" />
              <div className="card-body mt-3">
                <h4 className='name-testi'>JHONY</h4>
                <p className="card-text text-center">&quot;The game here is very unique, I usually play it with my child&quot;</p>
              </div>
            </div>
          </div>
          <div className='col-md-4'>
            <div className="card-testi">
              <Image src={Man2} className="card-img-top" alt="man" />
              <div className="card-body mt-3">
                <h4 className='name-testi'>PABLO</h4>
                <p className="card-text text-center">&quot;The support here really helps me when I have problems and the game is very cool to play&quot;</p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  )
}

export default HeroLandingPage;