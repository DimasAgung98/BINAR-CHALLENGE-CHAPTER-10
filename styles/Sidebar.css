.list-menu {
    list-style: none;
    display: flex;
    justify-content: center;
}

.list-menu li {
    padding: 20px 20px;
    text-align: center;
}

.list-menu a {
    text-decoration: none;
    color: black;
}

.list-menu a:hover {
    color: red
}