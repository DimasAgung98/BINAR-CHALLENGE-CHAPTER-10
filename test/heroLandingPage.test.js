import renderer from 'react-test-renderer';
import { createRef } from 'react';
import * as TestUtils from 'react-dom/test-utils';
import { render, screen } from '@testing-library/react';
import ReactDOM from 'react-dom/client';
import HeroLandingPage from '../components/landingPage/HeroLandingPage';
import { act } from 'react-dom/test-utils';
import Sidebar from '../components/HomePage/Sidebar';
import gambar from '../public/apex.jpg';
import Image from 'next/image';

describe('Testing Component HeroLandingPage', () => {
    test('Add Text HerolandingPage', () => {
        const component = renderer.create(
            <>
                <div>
                    <h1>Team One</h1>
                    <p>lorem ipsum</p>
                </div>
            </>
        )
        let page = component.toJSON();
        expect(page).toMatchSnapshot();
    });

    test('Add Form HerolandingPage', () => {
        const component = renderer.create(
            <>
                <div>
                    <h1>Team One</h1>
                </div>
                <div>
                    <form>Team :</form>
                    <input>One</input>
                </div>
            </>
        )
        let page = component.toJSON();
        expect(page).toMatchSnapshot();
    });

    test('Add Image HerolandingPage', () => {
        const component = renderer.create(
            <>
                <div>
                    <Image src={gambar} alt='image'></Image>
                </div>
            </>
        )
        let page = component.toJSON();
        expect(page).toMatchSnapshot();
    });

    let container;
    test('Add Componment sidebar HerolandingPage', () => {
        act(() => {
            ReactDOM.render(<Sidebar />);
        })
    });

    test('Expect Value HerolandingPage', () => {
        expect(HeroLandingPage()).toBe('teamone');
    });
})