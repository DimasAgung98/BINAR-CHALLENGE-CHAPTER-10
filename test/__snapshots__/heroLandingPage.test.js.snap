// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Component HeroLandingPage Add Form HerolandingPage 1`] = `
[
  <div>
    <h1>
      Team One
    </h1>
  </div>,
  <div>
    <form>
      Team :
    </form>
    <input>
      One
    </input>
  </div>,
]
`;

exports[`Testing Component HeroLandingPage Add Image HerolandingPage 1`] = `
<div>
  <span
    style={
      {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271%27%20height=%271%27/%3e"
        style={
          {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="image"
      data-nimg="intrinsic"
      decoding="async"
      onError={[Function]}
      onLoad={[Function]}
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="image"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src=""
        style={
          {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
</div>
`;

exports[`Testing Component HeroLandingPage Add Text HerolandingPage 1`] = `
<div>
  <h1>
    Team One
  </h1>
  <p>
    lorem ipsum
  </p>
</div>
`;
